src/middlewares/auth.js

```diff
@@ -15,4 +15,31 @@
         }
     });
 
+
+
+
+    var credientials;
+    // this first middleware sets the useCredientials attribute to empty, so
+    // adapters cannot use it directly.
+    exports.$Credientials = mw.Middleware(mw.$$Attr('useCredientials', function(args){
+      // Assign value for later checking
+      credientials = args.useCredientials
+
+      // Needs to return non-null and not-undefined
+      // in order for value to be (un)set
+      return '';
+    })).and(mw.$$Attr('credientials', function(args){
+        // check credientials for valid options, valid for fetch
```

Comment:
Can we narrow this down to just one attribute (`credentials`) and eliminate `withCredentias` (which is really just `credentials='include'`, right?)

That would simplify the user-facing API to be consistent across all adapters.